packages/uploadthing/src/h3.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  createRouter,
+  defineEventHandler,
+  getRequestHeaders,
+  getRequestURL,
+  readBody,
+  setHeaders,
+  setResponseStatus,
+} from "h3";
+
+import { getStatusCodeFromError, UploadThingError } from "@uploadthing/shared";
+
+import { UPLOADTHING_VERSION } from "./constants";
+import { defaultErrorFormatter } from "./internal/error-formatter";
+import {
+  buildPermissionsInfoHandler,
+  buildRequestHandler,
+} from "./internal/handler";
+import type { RouterWithConfig } from "./internal/handler";
+import type { FileRouter } from "./internal/types";
+
+export const createH3RouteHandler = <TRouter extends FileRouter>(
+  opts: RouterWithConfig<TRouter>,
+) => {
+  const requestHandler = buildRequestHandler(opts);
+  const POST = defineEventHandler(async (event) => {
+    const response = await requestHandler({
+      req: {
+        url: getRequestURL(event).href,
+        headers: getRequestHeaders(event),
+        json: () => Promise.resolve(readBody(event)),
+      },
+      //   res: event.node.res, // <-- DO WE NEED THIS?
```

Comment:
do we need to forward this for the middleware function to be able to run auth functions?

```ts
.middleware(({ req, res }) => {
  const session = await auth(req, res); // next-auth ish
})
```

Reply:
we'll pass the entire event instead